cmd/nodemon/nodemon.go

```diff
@@ -129,34 +132,27 @@ func run() error {
 
 	alerts := analyzer.Start(notifications)
 
-	socket, err := messaging.StartMessagingServer(nanomsgURL)
+	pubSubSocket, err := pubsub.StartPubSubMessagingServer(ctx, nanomsgPubSubURL, alerts)
+	if err != nil {
+		log.Printf("Failed to start pub sub messaging server: %v", err)
+		return err
+	}
+	defer func(socketPubSub protocol.Socket) {
+		if err := socketPubSub.Close(); err != nil {
+			log.Printf("Failed to close pubsub socket: %v", err)
+		}
+	}(pubSubSocket)
```

Comment:
I think `StartPubSubMessagingServer` function should be responsible to close this socket pair.